configure.ac

```diff
@@ -159,6 +159,35 @@ AC_PROG_CXX dnl required even with --disable-cxx due to automake conditionals
 m4_ifdef([AM_PROG_AR],[AM_PROG_AR])
 AC_PATH_PROG([PERL],[perl])
 
+# start gnu11/gnu99 check
+ORIG_CFLAGS=$CFLAGS
+WRAPPER_CC=$CC
+AC_SUBST(WRAPPER_CC)
+
+AC_MSG_CHECKING([if -std=gnu11 works])
+
+AC_LANG_PUSH([C])
+CFLAGS="-pedantic-errors -std=gnu11"
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+                   void g(int i){}
+                   #define f(X) _Generic((X), default: g)(X)
+                   ]], [[
+                        f(1);
+                        ]]
+                   )],
+                   [shmem_cv_c11_works="yes"], [shmem_cv_c11_works="no"])
+AS_IF([test "$shmem_cv_c11_works" = "no"],
+      [AC_MSG_RESULT([no])
+      AC_MSG_WARN([No C11 support detected, unable to test _Generic bindings.])
+      CFLAGS="-std=gnu99 $ORIG_CFLAGS"
```

Comment:
Do we want to use ```AC_PROG_CC_C99``` here instead of ```-std=gnu99```?  Not all C99 compilers support gnu99 or the ```-std``` argument.